hoptimator-jdbc/src/main/java/com/linkedin/hoptimator/jdbc/HoptimatorConnection.java

```diff
@@ -28,6 +32,26 @@ public HoptimatorConnection(CalciteConnection connection, Properties connectionP
     this.connectionProperties = connectionProperties;
   }
 
+  public ResolvedTable resolve(List<String> tablePath, Map<String, String> hints) throws SQLException {
+    try {
+      String tableSql = "SELECT * FROM " + tablePath.stream()
+          .map(x -> "\"" + x + "\"")
+          .collect(Collectors.joining("."));
+      RelNode tableRel = HoptimatorDriver.convert(this, tableSql).root.rel;
+      String namespace = "com.linkedin."
+            + tablePath.stream()
+            .map(x -> x.toLowerCase(Locale.ROOT))
+            .limit(tablePath.size() - 1)
+            .collect(Collectors.joining("."));
+      String schemaName = tablePath.get(tablePath.size() - 1).toLowerCase(Locale.ROOT);
+      org.apache.avro.Schema avroSchema = AvroConverter.avro(namespace, schemaName, tableRel.getRowType());
+      // TODO: generate source and sink configs via ConnectionService
+      return new ResolvedTable(tablePath, avroSchema, Collections.emptyMap(), Collections.emptyMap());
```

Comment:
Is the intention to provide the configs as if the tablePath would be used as a source or sink? And then whoever is using the table resolver will select the one they need?

Reply:
Yes. From the perspective of the data plane (e.g. a Flink catalog) we don't know how a table will be used until it is used. So we resolve for both situations, and the data plane can choose one or both as needed.

The connector configs may use the same or different physical connectors. For example, table templates could construct a table that reads with Iceberg and writes with Kafka. Theoretically, the source connector may itself be "hybrid", e.g. it could start reading from Iceberg and cutover to Kafka.

Reply:
Sounds reasonable to me. Should be pretty easy to add the ConnectionService call too. Might get a little awkward because some schemas can be used as a source and not a sink and maybe vice versa but as long as it's obvious to the user (empty map or null?) it should be fine.